prow/pjutil/trigger.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pjutil
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	pjapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	prowconfig "k8s.io/test-infra/prow/config"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/gcsupload"
+	"k8s.io/test-infra/prow/pod-utils/downwardapi"
+)
+
+// getJobArtifactsURL returns the artifacts URL for the given job
+func getJobArtifactsURL(prowJob *pjapi.ProwJob, config *prowconfig.Config) string {
+	var identifier string
+	if prowJob.Spec.Refs != nil {
+		identifier = fmt.Sprintf("%s/%s", prowJob.Spec.Refs.Org, prowJob.Spec.Refs.Repo)
+	} else {
+		identifier = fmt.Sprintf("%s/%s", prowJob.Spec.ExtraRefs[0].Org, prowJob.Spec.ExtraRefs[0].Repo)
+	}
+	spec := downwardapi.NewJobSpec(prowJob.Spec, prowJob.Status.BuildID, prowJob.Name)
+	jobBasePath, _, _ := gcsupload.PathsForJob(config.Plank.GetDefaultDecorationConfigs(identifier).GCSConfiguration, &spec, "")
+	return fmt.Sprintf("%s%s/%s",
+		config.Deck.Spyglass.GCSBrowserPrefix,
+		config.Plank.GetDefaultDecorationConfigs(identifier).GCSConfiguration.Bucket,
+		jobBasePath,
+	)
+}
+
+// Calls toJSON method on a jobResult type and writes it to the output path
+func writeResultOutput(prowjobResult JobResult, outputPath string, fileSystem afero.Fs) error {
```

Comment:
You can save yourself the interface definition and implementation by just accepting an empty interface (`interface{}`) here, just like `json.Marshal` does.

Also, I would suggest to yaml marshal this instead for readability?